src/http/parse_url.cpp

```diff
@@ -2,17 +2,17 @@
 // Measurement-kit is free software. See AUTHORS and LICENSE for more
 // information on the copying conditions.
 
-#include <measurement_kit/http.hpp>
 #include "src/ext/http-parser/http_parser.h"
+#include <measurement_kit/http.hpp>
 
 namespace mk {
 namespace http {
 
 Url parse_url(std::string url) {
     Url retval;
     http_parser_url url_parser;
-    http_parser_url_init(&url_parser); 
-    if (http_parser_parse_url(url.c_str(), url.length(), 0, &url_parser) != 0) {
+    http_parser_url_init(&url_parser);
+    if (http_parser_parse_url(url.data(), url.size(), 0, &url_parser) != 0) {
```

Comment:
Use my preferred C-string access functions (the `C++11` ones; more consistent, imho)